src/main/java/codeu/controller/ProfilePageServlet.java

```diff
@@ -52,7 +53,29 @@ public void init() throws ServletException {
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response)
             throws IOException, ServletException {
+
+        String requestUrl = request.getRequestURI();
+        String userTitle = requestUrl.substring("/profile/".length());
+
+
+        // Otherwise, they are looking for the profile of a particular user
+        // We will render the profile page for said user, given by profile.jsp (as opposed to profiles.jsp)
+        User user = userStore.getUser(userTitle);
+
+        if (user == null){
+            // cannot find user with name, redirect to homepage:
+            System.out.println("User was null: " + userTitle);
+            response.sendRedirect("/");
+        }
+
+        request.setAttribute("user", user);
         request.setAttribute("users", UserStore.getInstance().getUsersList());
-        request.getRequestDispatcher("/WEB-INF/view/profiles.jsp").forward(request, response);
+        request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
+    
+        
```

Comment:
You probably should fix the style and the extra spaces in here. Also, the line you deleted and added here was the same, I don't know if you forgot to change the path or it was an accident

Reply:
oh no it's changed slightly. Notice one says profiles.jsp, the other is profile.jsp. They deliver different things.

Reply:
Also what do you mean extra spaces? Are you referring to the line or just the styling of this file in general?

Reply:
Yeah just make sure there aren't all those extra lines in the code. Oops, my bad I didn't catch that, then all good!